py-polars/polars/internals/expr/categorical.py

```diff
@@ -1,16 +1,24 @@
 from __future__ import annotations
 
+import sys
 from typing import TYPE_CHECKING
 
 import polars.internals as pli
 
 if TYPE_CHECKING:
     from polars.internals.type_aliases import CategoricalOrdering
 
+    if sys.version_info >= (3, 8):
```

Comment:
Could we do this one level higher? So that every module below import `Final` and this branch with checking the `sys.version` can only be done once. I see we do this check in 9 files.

Reply:
I don't like the idea of re-exporting typing imports, but if this bothers you we can simply get rid of the `Final` designation on the accessor variables. Then they will just be strings. I'll add a commit for this.

Reply:
> > > This all looks very interesting and it seems really clean to be able to dispatch this all with a decorator. Thanks for all the help op this. I only do want to be certain that we do not break autocomplete/ parameter hinting in:
> > > 
> > > * [x]    vscode
> > > * [x]    pycharm/jetbrains
> > > * [x]    jupyter notebooks/ ipython
> > 
> > 
> > I can confirm that PyCharm/DataGrip (Jetbrains) are in good shape, along with `.ipynb` running under PyCharm.
> 
> Everything looks good in VSCode, as far as I can tell!

Alright good to hear. Then here we go. :shipit: :)